agent/test-agent/src/k8s_client.rs

```diff
@@ -129,4 +152,9 @@ impl Client for DefaultClient {
     async fn results_file(&self) -> Result<PathBuf, Self::E> {
         return Ok(PathBuf::from(TESTSYS_RESULTS_FILE));
     }
+
+    async fn get_test_status(&self) -> Result<TestStatus, Self::E> {
```

Comment:
This can be done with `spec()`

Reply:
Spec doesn't contain any functionality to get status, hence a new function has been created to get status from test CRD

Reply:
Oh, I see. Let's just call it `test_status` to keep naming consistent.